documentation/src/main/style/asciidoctor/js/toc.js

```diff
@@ -1,32 +1,66 @@
-var versions = {
-    'current' : '/current/userguide/html_single/Hibernate_User_Guide.html',
-    '6.1' : '/6.1/userguide/html_single/Hibernate_User_Guide.html',
-    '6.0' : '/6.0/userguide/html_single/Hibernate_User_Guide.html',
-    '5.6' : '/5.6/userguide/html_single/Hibernate_User_Guide.html',
-    '5.5' : '/5.5/userguide/html_single/Hibernate_User_Guide.html',
-    '5.4' : '/5.4/userguide/html_single/Hibernate_User_Guide.html',
-    '5.3' : '/5.3/userguide/html_single/Hibernate_User_Guide.html',
-    '5.2' : '/5.2/userguide/html_single/Hibernate_User_Guide.html',
-    '5.1' : '/5.1/userguide/html_single/Hibernate_User_Guide.html',
-    '5.0' : '/5.0/userguide/html_single/Hibernate_User_Guide.html',
-    '4.3' : '/4.3/manual/en-US/html_single/',
-    '4.2' : '/4.2/manual/en-US/html_single/',
-    '4.1' : '/4.1/manual/en-US/html_single/',
-    '4.0' : '/4.0/manual/en-US/html_single/',
-    '3.6' : '/3.6/reference/en-US/html_single/',
-    '3.5' : '/3.5/reference/en-US/html_single/',
-    '3.3' : '/3.3/reference/en-US/html_single/',
-    '3.2' : '/3.2/reference/en/html_single/'
-};
+var versions = [
+    // new versions will be dynamically populated by web scraping directory page
+    '6.6', '6.5', '6.4', '6.3', '6.2', '6.1', '6.0', // 6.x series
+    '5.6', '5.5', '5.4', '5.3', '5.2', '5.1', '5.0', // 5.x series
+    '4.3', '4.2', '4.1', '4.0',                      // 4.x series
+    '3.6', '3.5', '3.4', '3.3', '3.2'                // 3.x series
+];
```

Comment:
to minimize risk, the above static version will be there always, so in event of ajax dynamic logic breaks, we won't lose much.

The dynamic logic will manifest itself since v7.x. 

I tested it suceesuflly in chrome and managed to use jQuery functions always, so in theory it should work in all browser.

Reply:
We might well need some front end developer to reivew the logic here.